### PR TITLE
Hopper

### PR DESCRIPTION
**Pull request by AndreasLc1103**
Added Ball Hopper Subsystem

### CODE REVIEW

File: src/com/nutrons/steamworks/Hopper.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+public class Hopper implements Subsystem {
+    private final Flowable<ControllerEvent> runHopper;
+    private final Consumer<ControllerEvent> hopperController;
+
+    /**
+     * HopperController
+     * Tune RunAtPowerEvent
+     */
+    public Hopper(Consumer<ControllerEvent> HopperController) {
+        this.runHopper = Flowable.just(new RunAtPowerEvent(0.9));
```
Comment (laposata12) [inline src/com/nutrons/steamworks/Hopper.java]: the 0.9 should be a variable rather than a constant

File: src/com/nutrons/steamworks/Hopper.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import org.reactivestreams.Subscriber;
+
+
+public class Hopper implements Subsystem {
+    private final Flowable<ControllerEvent> runHopper;
+    private final Consumer<ControllerEvent> hopperController;
+<<<<<<< HEAD
+    private static double spinsPower = 0.9;
+
+=======
```
Comment (kreeve) [inline src/com/nutrons/steamworks/Hopper.java]: Get rid of merge conflicts!

**Review by kreeve — DISMISSED** [review]

File: src/com/nutrons/steamworks/RobotMap.java [hunk 3]
```diff
@@ -7,5 +7,6 @@
     public static final int INTAKE_MOTOR = 4;
     public static final int SHOOTER_MOTOR_1 = 2;
     public static final int SHOOTER_MOTOR_2 = 3;
-
+    public static final int TOP_HOPPER_MOTOR = 0; // need to change ln 10 and 11 when we get actual port #
```
Comment (MiloDavis) [inline src/com/nutrons/steamworks/RobotMap.java]: Make into TODO

File: src/com/nutrons/steamworks/Hopper.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import io.reactivex.functions.Consumer;
+import org.reactivestreams.Subscriber;
+
+
+public class Hopper implements Subsystem {
+    private final Flowable<ControllerEvent> runHopper;
+    private final Consumer<ControllerEvent> hopperController;
+    private static double spinsPower = 0.9;
+    /**
+     * HopperController
```
Comment (MiloDavis) [inline src/com/nutrons/steamworks/Hopper.java]: Make into real [javadoc](http://www.drjava.org/docs/user/ch10.html)

File: src/com/nutrons/steamworks/Hopper.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import com.nutrons.framework.controllers.Talon;
+import io.reactivex.Flowable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.functions.Consumer;
+import org.reactivestreams.Subscriber;
+
+
+public class Hopper implements Subsystem {
+    private final Flowable<ControllerEvent> runHopper;
+    private final Consumer<ControllerEvent> hopperController;
```
Comment (MiloDavis) [inline src/com/nutrons/steamworks/Hopper.java]: Comment about this being a leader and follower

**Review by MiloDavis — CHANGES_REQUESTED** [review]

File: src/com/nutrons/steamworks/Hopper.java (hunk 4 above)
Comment (MiloDavis) [reply src/com/nutrons/steamworks/Hopper.java]: This is a constant, mark it as such make `final`

File: src/com/nutrons/steamworks/RobotBootstrapper.java [hunk 6]
```diff
@@ -10,9 +10,13 @@
     private WpiXboxGamepad controller;
     private Talon shooterMotor1;
     private Talon shooterMotor2;
+    private Talon TopHopperMotor;
```
Comment (MiloDavis) [inline src/com/nutrons/steamworks/RobotBootstrapper.java]: Lowercase variable names

**Review by MiloDavis — CHANGES_REQUESTED** [review]

File: src/com/nutrons/steamworks/Hopper.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import com.nutrons.framework.controllers.RunAtPowerEvent;
+import com.nutrons.framework.controllers.Talon;
+import io.reactivex.Flowable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.functions.Consumer;
+import org.reactivestreams.Subscriber;
+
+
+public class Hopper implements Subsystem {
+    private final static double spinsPower = 0.9;
```
Comment (MiloDavis) [inline src/com/nutrons/steamworks/Hopper.java]: Constant notation `SPIN_POWER`

File: src/com/nutrons/steamworks/RobotMap.java [hunk 8]
```diff
@@ -7,5 +7,7 @@
     public static final int INTAKE_MOTOR = 4;
     public static final int SHOOTER_MOTOR_1 = 2;
     public static final int SHOOTER_MOTOR_2 = 3;
-
+    // @TODO Change Assign hopper ports when given robot
```
Comment (MiloDavis) [inline src/com/nutrons/steamworks/RobotMap.java]: `// TODO: Change hopper ports to match robot`

**Review by MiloDavis — CHANGES_REQUESTED** [review]

**Review by MiloDavis — APPROVED** [review]

**Pull request merged**